pkg/controller/appdefinition/jobs.go

```diff
@@ -80,6 +80,10 @@ func toJob(req router.Request, appInstance *v1.AppInstance, pullSecrets *PullSec
 		},
 	}
 
+	if perms := v1.FindPermission(name, appInstance.Spec.Permissions); perms.HasRules() {
+		jobSpec.Template.Spec.ServiceAccountName = name
```

Comment:
Hmm.... this won't work.  We need the service accounts to always exist even if there is no permission assigned to them.  The is important for cloud IAM authorizations.  I realize this is probably going to change the logic quite a bit because if the service account object itself doesn't exist the pod won't start, so we have to ensure SA objects alway get created.

Reply:
The most recent push should address your comment here. See my comment on the other one you left for more information.